### PR TITLE
TS-4548: Convert custom log config file to Lua

### PR DESCRIPTION
**Pull request by jpeach**
Initial implementation of ``logging.config`` which is a Lua configuration file for custom logs.

### CODE REVIEW

**Comment by jpeach** [comment]
Ping @shukitchan @zwoop

**Comment by zwoop** [comment]
Ding dong, XML is dead? :-)

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/518/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/415/ for details.

**Comment by shukitchan** [comment]
Yeah. That's indeed a good thing to finally kill off XML. I will give some time to review it today/tomorrow.

**Comment by zwoop** [comment]
Fwiw, it's running on docs/ci, with the slightly modified squid log that we use there.

File: proxy/logging/LogConfig.cc [hunk 1]
```diff
@@ -493,14 +494,21 @@ void
 LogConfig::setup_log_objects()
 {
   Debug("log", "creating objects...");
+  RecBool enable_lua = false;
+
+  RecGetRecordBool("proxy.config.log.config.enable_lua", &enable_lua);
```
Comment (zwoop) [inline proxy/logging/LogConfig.cc]: I think you should nuke this config, and just make the new config default.

**Comment by shukitchan** [comment]
👍 

Good one. I can now have an string array of host names. Loop through them and create log object for each of them that is slightly different from one another.

File: proxy/logging/LogConfig.cc (hunk 1 above)
Comment (jpeach) [inline proxy/logging/LogConfig.cc]: Would prefer to do that as a separate Jira.

**Comment by jpeach** [comment]
@zwoop @shukitchan I removed the configuration option and added a little Lua stack helper class I had been thinking about.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/535/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/432/ for details.

**Pull request merged**